meson.build

```diff
@@ -389,22 +389,27 @@ luaold_interpreters = [
 
 lua_pc_name = ''
 
-if sys_windows or get_option('lua-interpreter') == 'lua'
-  if sys_windows and get_option('lua-interpreter') != 'lua'
-    warning('There is no "' + get_option('lua-interpreter')
-            + '" binding option for Windows, backing up to "lua"')
-  endif
-  config_h.set('ENABLE_LUA_OLD', '1')
-  foreach l : luaold_interpreters
-    lua = dependency(l[0], version: l[1], required:false)
-    lua_pc_name = l[0]
-    if lua.found() == true
-      break
+if not lua.found():  # see header_checks/meson.build
```

Comment:
This will cause an error on Linux, since `lua` was never defined.

Reply:
Also, we're changing the behaviour here. What if our backend is actually luajit?